study-builder/src/main/java/org/broadinstitute/ddp/studybuilder/task/MBCAboutYouV2.java

```diff
@@ -0,0 +1,139 @@
+        gson = GsonUtil.standardGson();
+
+    }
+
+    @Override
+    public void run(Handle handle) {
+        //creates version: 2 for AboutYou activity.
+        //Add updated new RACE question and disable existing RACE and HISPANIC questions
+        //Add GENDER and SEX questions
+
+        LanguageStore.init(handle);
+        User adminUser = handle.attach(UserDao.class).findUserByGuid(studyCfg.getString("adminUser.guid")).get();
+        StudyDto studyDto = handle.attach(JdbiUmbrellaStudy.class).findByStudyGuid(studyCfg.getString("study.guid"));
+        var activityDao = handle.attach(ActivityDao.class);
+        String activityCode = dataCfg.getString("activityCode");
+        String studyGuid = studyDto.getGuid();
+        long studyId = studyDto.getId();
+        long activityId = ActivityBuilder.findActivityId(handle, studyId, activityCode);
+        SqlHelper helper = handle.attach(SqlHelper.class);
+        String reason = String.format(
+                "Update activity with studyGuid=%s activityCode=%s to versionTag=%s",
+                studyGuid, activityCode, versionTag);
+        RevisionMetadata meta = new RevisionMetadata(timestamp.toEpochMilli(), adminUser.getId(), reason);
+
+        JdbiRevision jdbiRevision = handle.attach(JdbiRevision.class);
+
+        //change version
+        ActivityVersionDto activityVersionDto = activityDao.changeVersion(activityId, versionTag, meta);
+
+        //disable RACE question
+        JdbiQuestion jdbiQuestion = handle.attach(JdbiQuestion.class);
+        JdbiFormSectionBlock jdbiFormSectionBlock = handle.attach(JdbiFormSectionBlock.class);
+        QuestionDao questionDao = handle.attach(QuestionDao.class);
+
+        QuestionDto currRaceDto = jdbiQuestion.findLatestDtoByStudyIdAndQuestionStableId(studyId, "RACE").get();
+        long terminatedRevId = jdbiRevision.copyAndTerminate(currRaceDto.getRevisionId(), meta);
+        long currRaceBlockId = helper.findQuestionBlockId(currRaceDto.getId());
+        SectionBlockMembershipDto currRaceSectionDto = jdbiFormSectionBlock.getActiveMembershipByBlockId(currRaceBlockId).get();
+        helper.updateFormSectionBlockRevision(currRaceBlockId, terminatedRevId);
+
+        //disable HISPANIC question
+        QuestionDto currHispanicDto = jdbiQuestion.findLatestDtoByStudyIdAndQuestionStableId(studyId, "HISPANIC").get();
+        long currHispanicBlockId = helper.findQuestionBlockId(currHispanicDto.getId());
+        questionDao.disablePicklistQuestion(currHispanicDto.getId(), meta);
+        helper.updateFormSectionBlockRevision(currHispanicBlockId, terminatedRevId);
+
+        //add new RACE (self) question
+        SectionBlockDao sectionBlockDao = handle.attach(SectionBlockDao.class);
+        FormBlockDef raceDef = gson.fromJson(ConfigUtil.toJson(dataCfg.getConfig("raceQuestion")), FormBlockDef.class);
+        long newV2RevId = jdbiRevision.insertStart(timestamp.toEpochMilli(), adminUser.getId(), "MBC aboutyou version#2");
```

Comment:
The v2 `ActivityVersionDto` has a rev id. I think we can use that rev id instead of creating a new one. And this ensures that the timestamps there will definitely match up.